src/Moryx.Runtime.Kernel/FileSystem/LocalFileSystem.cs

```diff
@@ -0,0 +1,278 @@
+﻿using Castle.MicroKernel.Registration;
+using Microsoft.Extensions.Logging;
+using Moryx.FileSystem;
+using Moryx.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Moryx.Runtime.Kernel.FileSystem
+{
+    internal class LocalFileSystem : IMoryxFileSystem
+    {
+        private string _fsDirectory;
+        private string _ownerFilesDirectory;
+        private readonly ILogger _logger;
+
+        private readonly List<MoryxFileTree> _ownerTrees = new List<MoryxFileTree>();
+
+        public LocalFileSystem(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger(nameof(LocalFileSystem));
+        }
+
+        public void SetBasePath(string basePath = "fs")
```

Comment:
Isn't this somehow duplicated in the service collection extension `UseMoryxFileSystem`? Also it only allows for directories within the apps directory which is not recommended by .net 🙈

Reply:
Well the extension calls the method. Technically I might work with currentDir/../file-system.

However we could also use an absolute path.

Reply:
Ohh sorry, I apparently wasn't fully awake yet 🙈 

I think absolute or `currentDir/../file-system` could work. I guess checking the user input would be a good thing and then either taking the absulute patch given or creating the directory with the relative path 🤔